### PR TITLE
5 Unfinity cards + clown token + Fix + AI improvements

### PR DESCRIPTION
**Pull request by Simisays**

### CODE REVIEW

File: forge-gui/res/cardsfolder/upcoming/six_sided_die.txt [hunk 1]
```diff
@@ -0,0 +1,14 @@
+Name:Six-Sided Die
+ManaCost:2 B
+Types:Instant
+A:SP$ Pump | ValidTgts$ Creature | SubAbility$ DBRollDice | IsCurse$ True | SpellDescription$ Choose target creature. Roll a six-sided die
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/six_sided_die.txt:4]: `die` > `die.`
Needs a StackDescription

File: forge-gui/res/cardsfolder/upcoming/vegetation_abomination.txt [hunk 2]
```diff
@@ -0,0 +1,9 @@
+Name:Vegetation Abomination
+ManaCost:2 G
+Types:Artifact Creature Plant Mutant Food
+PT:2/2
+K:Deathtouch
+A:AB$ RollDice | Cost$ T Sac<1/CARDNAME> | ResultSVar$ Result | SubAbility$ DBGainLife | SpellDescription$ Roll a six-sided die. You gain life equal to the result.
+SVar:DBGainLife:DB$ GainLife | LifeAmount$ Result
+DeckHas:Ability$Lifegain|Sacrifice
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/vegetation_abomination.txt:8]: LifeGain

File: forge-gui/res/cardsfolder/b/bad_moon.txt [hunk 3]
```diff
@@ -3,4 +3,7 @@ ManaCost:1 B
 Types:Enchantment
 S:Mode$ Continuous | Affected$ Creature.Black | AddPower$ 1 | AddToughness$ 1 | Description$ Black creatures get +1/+1.
 SVar:PlayMain1:TRUE
+SVar:NeedsToPlayVar:CountOpps LTCountMe
+SVar:CountOpp:Count$Valid Creature.OppCtrl+Black/LimitMax.5
+SVar:CountMe:Count$Valid Creature.YouCtrl+Black
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/b/bad_moon.txt:8]: Should have DeckNeeds for Black color
below for White

File: forge-gui/res/cardsfolder/k/karma.txt [hunk 4]
```diff
@@ -4,5 +4,7 @@ Types:Enchantment
 T:Mode$ Phase | Phase$ Upkeep | ValidPlayer$ Player | TriggerZones$ Battlefield | Execute$ TrigDamage | TriggerDescription$ At the beginning of each player's upkeep, CARDNAME deals damage to that player equal to the number of Swamps they control.
 SVar:TrigDamage:DB$ DealDamage | Defined$ TriggeredPlayer | NumDmg$ X
 SVar:X:Count$Valid Swamp.ActivePlayerCtrl
-AI:RemoveDeck:Random
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/k/karma.txt:7]: the AI tag should be kept, when generating random deck AI can't know what cards opponent will have
same for below

**Review by tool4ever — APPROVED** [review]

**Pull request merged**